tensorboard/tools/whitespace_hygiene_test.py

```diff
@@ -0,0 +1,135 @@
+    "tensorboard/plugins/hparams/tf_hparams_parallel_coords_plot/tf-hparams-parallel-coords-plot.html",
+    "tensorboard/plugins/hparams/tf_hparams_sessions_pane/tf-hparams-sessions-pane.html",
+    "tensorboard/plugins/interactive_inference/witwidget/BUILD",
+    "tensorboard/plugins/interactive_inference/witwidget/pip_package/BUILD",
+    "tensorboard/plugins/interactive_inference/witwidget/pip_package/RELEASE.md",
+    "tensorboard/plugins/mesh/BUILD",
+    "tensorboard/plugins/mesh/README.md",
+    "tensorboard/plugins/mesh/http_api.md",
+    "tensorboard/plugins/mesh/mesh_plugin_test.py",
+    "tensorboard/plugins/mesh/metadata_test.py",
+    "tensorboard/plugins/mesh/summary_test.py",
+    "tensorboard/plugins/mesh/test_utils.py",
+    "tensorboard/plugins/pr_curve/http_api.md",
+    "tensorboard/plugins/profile/README.md",
+    "tensorboard/plugins/profile/pod_viewer/stack_bar_chart/stack-bar-chart.ts",
+    "tensorboard/plugins/projector/vz_projector/umap.d.ts",
+    "tensorboard/plugins/projector/vz_projector/util.ts",
+    "tensorboard/plugins/projector/vz_projector/vz-projector-inspector-panel.ts",
+    "tensorboard/plugins/projector/vz_projector/vz-projector-metadata-card.html",
+    "tensorboard/plugins/projector/vz_projector/vz-projector-projections-panel.html",
+    "tensorboard/plugins/projector/vz_projector/vz-projector-projections-panel.ts",
+    "tensorboard/plugins/projector/vz_projector/vz-projector.ts",
+    "tensorboard/program_test.py",
+    "tensorboard/tools/docs_list_format_test.sh",
+])
+
+
+Match = collections.namedtuple("Match", ("filename", "line_number", "line"))
+
+
+def main():
+  chdir_to_repo_root()
+  matches = git_grep("  *$")
+  errors = [m for m in matches if m.filename not in exceptions]
+  okay = True
+
+  if errors:
+    print("Superfluous trailing whitespace:")
+    for error in errors:
+      print("%s:%d:%s$" % (error.filename, error.line_number, error.line))
+    print()
+    okay = False
+
+  stale_exceptions = exceptions - frozenset(m.filename for m in matches)
+  if stale_exceptions:
+    print("Stale exceptions (no whitespace problems; prune exceptions list):")
+    for filename in stale_exceptions:
+      print(filename)
+    print()
+    okay = False
+
+  sys.exit(0 if okay else 1)
+
+
+def git_grep(pattern):
+  """Run `git grep` and collect matches.
+
+  This function exits the process if `git grep` writes any stderr: for
+  instance, if the provided pattern is an invalid regular expression.
+
+  Args:
+    pattern: `str`; a pattern argument to `git grep`.
+
+  Returns:
+    A list of `Match` values.
+  """
+  cmd = ["git", "grep", "-Izn", "--", pattern]
+  p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  (stdout, stderr) = p.communicate()
+  if stderr:
+    getattr(sys.stderr, "buffer", sys.stderr).write(stderr)  # Python 2 compat
+    sys.exit(1)
+  result = []
+  while stdout:
+    (filename_raw, _unused_nul, rest) = stdout.partition(b"\0")
```

Comment:
Since realistically we're not going to have newlines in filenames committed to our git repo, it seems simpler and more obvious to just do `for line in stdout.splitlines(): filename_raw, line_number_raw, line_number = line.split(b"\0", 2)`.

Reply:
IMHO, the correct version is sufficiently simple as to be
unobjectionable. But in the interest of compromise, I can switch this
over to the incorrect-but-fewer-lines version. (I’d rather cede this
point, which is under our control, than the other, which is under the
control of whoever clones the repository.)